### PR TITLE
Sf Tab Navigation

### PR DESCRIPTION
**Pull request by RakowskiPrzemyslaw**
# Related issue
https://github.com/DivanteLtd/storefront-ui/issues/57

# Scope of work
API proposition for this solution
image

with custom navigation items
image

with custom navigation
image

# Screenshots of visual changes
image

with custom navigation items
image

# Checklist

- [ ] I followed [composition rules](https://github.com/DivanteLtd/storefront-ui/blob/master/docs/component-rules.md) for my component
- [ ] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))

### CODE REVIEW

**Comment by filrak** [comment]
API looks good to me! :) I'll dig into internals on Monday.

File: src/components/organisms/SfTabMenu/SfTab/SfTab.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+export const TAB_COMPONENT_NAME = "SfTab";
+
+export default {
+  name: TAB_COMPONENT_NAME,
+  inject: ["sharedState"],
```
Comment (filrak) [inline src/components/organisms/SfTabMenu/SfTab/SfTab.js]: use this.$parent or (even better) explict props from slot scope

Comment (RakowskiPrzemyslaw) [reply src/components/organisms/SfTabMenu/SfTab/SfTab.js]: without provide/inject?

Comment (filrak) [reply src/components/organisms/SfTabMenu/SfTab/SfTab.js]: injectors are just harder for people to use than scoped slots. If one wants to have custom nav i guess we shouldn't force them to use injectors

Comment (filrak) [reply src/components/organisms/SfTabMenu/SfTab/SfTab.js]: Nvmd, i rethinked and it's fine

**Comment by filrak** [comment]
As we discussed on priv - green light! ;)

**Comment by RakowskiPrzemyslaw** [comment]
problem with dynamic tabs solved in last commit, now you can dynamically change tabs and they will be registered, how could i solve problem with ssr?

**Comment by RakowskiPrzemyslaw** [comment]
now everything will work without providing 'id' which now is 'value', if not provided it will be set automatically , v-model support added

**Comment by filrak** [comment]
Also according to designs radiobox is on the top of content box. not in the middle.

File: src/components/organisms/SfTabMenu/SfTabs/SfTabs.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import SfTab, { TAB_COMPONENT_NAME } from "../SfTab/SfTab.vue";
+import SfTabsNavigation from "../SfTabsNavigation/SfTabsNavigation.vue";
+import Vue from "vue";
```
Comment (SebSzcze) [inline src/components/organisms/SfTabMenu/SfTabs/SfTabs.js]: unnecessary import

File: src/components/organisms/SfTabMenu/SfTabs/SfTabs.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    });
+  },
+
+  updated() {
+    this.updateTabs();
+  },
+
+  methods: {
+    changeTab(tab) {
+      Vue.set(this.sharedState, "activeTab", tab);
```
Comment (SebSzcze) [inline src/components/organisms/SfTabMenu/SfTabs/SfTabs.js]: `this.$set`

Comment (RakowskiPrzemyslaw) [reply src/components/organisms/SfTabMenu/SfTabs/SfTabs.js]: https://github.com/DivanteLtd/storefront-ui/pull/87/commits/abf8fc3d499d271a07f896b5d1bff1950003f743

File: src/components/organisms/SfTabMenu/SfTabs/SfTabs.js (hunk 2 above)
Comment (RakowskiPrzemyslaw) [reply src/components/organisms/SfTabMenu/SfTabs/SfTabs.js]: https://github.com/DivanteLtd/storefront-ui/pull/87/commits/abf8fc3d499d271a07f896b5d1bff1950003f743

**Comment by filrak** [comment]
discussed on priv.
Please make examples cleaner and better looking ;)

**Comment by leomp12** [comment]
Hey @RakowskiPrzemyslaw , are you planning to update this pr or I can do that?

**Comment by RakowskiPrzemyslaw** [comment]
Hi @leomp12, there are only few css tweaks to be made, I'll finish it this week

**Comment by leomp12** [comment]
Great @RakowskiPrzemyslaw , thanks!

**Pull request closed without merging**